lib/auth0verification.js

```diff
@@ -35,28 +36,29 @@ const getExcludedRoutes = function (configuredExclusions) {
 };
 
 const getAuthPublicKey = function (jwksUrl, cache, kid) {
-  cache.get(`kid${kid}`).then(function (data) {
-    if (!data) {
-      var client = jwksClient({
-        cache: true,
-        cacheMaxEntries: 5,
-        cacheMaxAge: ms('10h'),
-        JwksUri
-      });
-      client.getSigningKey(kid, (err, key) => {
-        if (err) {
-          return null;
-        } else {
-          var signingKey = key.publicKey || key.rsaPublicKey;
-          var encodedValue = new Buffer(signingKey).toString("base64");
-          cache.set(`kid${kid}`, encodedValue, 36000).then(() => {
-            return new Buffer(signingKey);
-          });
-        }
-      });
-    } else {
-      return new Buffer(data, "base64");
-    }
+  return new Promise((resolve, reject) => {

```

Comment:
I tried returning the cache.get without wrapping it in a promise but I saw the same behavior as before, the jwtClient returned after calling this function and failed